File: server/mc/common/src/main/java/calebxzhou/rdi/mc/common/WsMessage.java
package calebxzhou.rdi.mc.common;

/**
 * calebxzhou @ 2026-01-12 17:57
 */

public final class WsMessage  {
    private final int id;
    private final Channel channel;
    private final String data;

    public WsMessage(int id, Channel channel, String data) {
        this.id = id;
        this.channel = channel;
        this.data = data;
    }

    public int getId() {
        return id;
    }

    public Channel getChannel() {
        return channel;
    }

    public String getData() {
        return data;
    }

    public enum Channel {
        Command,
        Response
    }
}

File: server/master/src/main/kotlin/calebxzhou/rdi/master/model/WsMessage.kt
package calebxzhou.rdi.master.model

import kotlinx.serialization.Serializable


@Serializable
data class WsMessage<T>(
    val id: Int,
    val channel: Channel,
    val data: T,
) {

    @Serializable
    enum class Channel {
        Command,
        Response,
    }
}
